src/history/hs.c

```diff
@@ -799,29 +800,14 @@ __wt_hs_insert_updates(WT_SESSION_IMPL *session, WT_PAGE *page, WT_MULTI *multi)
                 continue;
 
             non_aborted_upd = upd;
-
-            /* If we've seen a smaller timestamp before, use that instead. */
-            if (min_insert_ts < upd->start_ts) {
```

Comment:
I'd rather this code stayed here, and handled WT_TS_NONE specially, than it is moved below.

I'm concered with the code movement that some (pathological) combination of mixed mode and out of order timestamps will mean that we don't fixup out-of-order timestamps.

Reply:
Understood.

Reply:
I think the logic should be we fix all the out of order timestamp before any mixed mode operation and ignore all the updates with timestamps after we see the first mixed mode operation. So I moved the logic there.

Reply:
Done